src/server/services/bountyEntry.service.ts

```diff
@@ -328,53 +328,90 @@ export const awardBountyEntry = async ({ id, userId }: { id: number; userId: num
 
       switch (updatedBenefactor.currency) {
         case Currency.BUZZ: {
-          if (updatedBenefactor.buzzTransactionId) {
-            // This is a multi-transaction
-            const data = await getMultiAccountTransactionsByPrefix(
-              updatedBenefactor.buzzTransactionId
+          if (
+            updatedBenefactor.buzzTransactionId &&
+            updatedBenefactor.buzzTransactionId.length > 0
+          ) {
+            // Collect all transactions from all transaction IDs (batch optimization)
+            const allTransactions = [];
+            let totalAmount = 0;
+
+            // Process all transaction IDs in parallel for better performance
+            const txResults = await Promise.allSettled(
+              updatedBenefactor.buzzTransactionId.map(async (txId) => {
+                const data = await getMultiAccountTransactionsByPrefix(txId);
+                const txAmount = data.reduce((sum, t) => sum + t.amount, 0);
```

Comment:
Data would have multi-color buzz, We shouldn't sum that out but split it accordinly into the right types.